client/src/js/app.js

```diff
@@ -108,8 +108,14 @@ function bhimaconfig($routeProvider) {
     templateUrl: '/partials/sales/sales.html'
   })
   .when('/sale_records/:recordID?', {
-    controller: 'salesRecords',
-    templateUrl: '/partials/records/sales_records/sales_records.html'
+    controller: 'SalesRecords as SalesRecordsCtrl',
+    templateUrl: '/partials/records/sales_records/sales_records.html',
+    resolve : { 
+      'ReportServiceData' : ['ReportService', function (ReportService) { 
```

Comment:
While I can definitely see the appeal of this method of injecting data, will we ever go to lazy-loaded routes?  In which case, it is hard to imagine how we'd port this code.